localstack/services/infra.py

```diff
@@ -304,6 +304,9 @@ def cleanup_resources():
     cleanup_tmp_files()
     cleanup_threads_and_processes()
 
+    if config.CLEAR_TMP_FOLDER:
+        files.rm_rf(config.dirs.tmp)
```

Comment:
Note to self: my first intuition was that we should potentially add some guards here against unwanted deletion of top-level folders if things are misconfigured (e.g., in host mode), but after following the logic above this should be safe 👌 (`config.dirs.tmp` can never point to problematic paths like `/` or similar)